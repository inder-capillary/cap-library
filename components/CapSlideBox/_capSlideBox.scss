@import "../styles/_variables.scss";

$slideBox: "cap-slide-box-v2";

.#{$slideBox}.show-shadow {
  .slidebox-header {
    -webkit-box-shadow: 0px 0px 4px 2px rgba(0, 0, 0, 0.12);
    -moz-box-shadow: 0px 0px 4px 2px rgba(0, 0, 0, 0.12);
    box-shadow: 0px 0px 4px 2px rgba(0, 0, 0, 0.12);
  }
  .slidebox-footer {
    -webkit-box-shadow: 0px 0px 4px 2px rgba(0, 0, 0, 0.12);
    -moz-box-shadow: 0px 0px 4px 2px rgba(0, 0, 0, 0.12);
    box-shadow: 0px 0px 4px 2px rgba(0, 0, 0, 0.12);
  }
}

.#{$slideBox} {
  width: 100vw;
  height: 100vh;
  position: fixed;
  top: 0;
  left: 0;
  background: rgba(0, 0, 0, 0.56);
  z-index: 100;

  &.hide-slidebox {
    visibility: hidden;
    opacity: 0;
    transition-delay: 0s;
  }
  &.show-slidebox {
    visibility: visible;
    opacity: 1;
    transition: visibility 1s linear 0.33s, opacity 0.33s linear;
  }
}

.#{$slideBox}-container {
  position: absolute;
  height: 100vh;
  background: rgba(255, 255, 255, 1);
  overflow-y: auto;
  .slidebox-header {
    padding: 0 $CAP_SPACE_48;
    justify-content: space-between;
    align-items: center;
    display: flex;
    height: 92px;
    position: relative;
    z-index: 1;
  }
  .slidebox-footer {
    padding: 0 $CAP_SPACE_48;
    position: absolute;
    bottom: 0;
    width: 100%;
    display: flex;
    align-items: center;
    height: 88px;
  }
  .slidebox-content-container {
    padding: 0 $CAP_SPACE_48;
    overflow-y: auto;
    height: calc(100vh - 92px);
    &.has-footer {
      height: calc(100vh - 92px - 88px);
    }
  }
}

.#{$slideBox}-container.size-r {
  min-width: 660px;
  max-width: 800px;
  width: 50vw;
}

.#{$slideBox}-container.size-s {
  min-width: 372px;
  max-width: 400px;
  width: 25vw;
  .slidebox-header {
    padding: 0 $CAP_SPACE_24;

  }
  .slidebox-footer {
    padding: 0 $CAP_SPACE_24;

  }
  .slidebox-content-container {
    padding: 0 $CAP_SPACE_24;

  }
}

.#{$slideBox}-container.size-l {
  min-width: 948px;
  max-width: 1140px;
  width: 75vw;
}

.#{$slideBox}-container.left {
  left: 0;
}

.#{$slideBox}-container.right {
  right: 0;
}

.#{$slideBox}-container.size-xl {
  min-width: 1180px;
  max-width: 1280px;
  width: 1180px;
}
