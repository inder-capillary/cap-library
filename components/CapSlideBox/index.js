/**
*
* CapSlideBox
*
*/

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { throttle } from 'lodash';
import CapHeading from '../CapHeading';
import CapIcon from '../CapIcon';

import './_capSlideBox.scss';


const clsPrefix = 'cap-slide-box-v2';

export default class CapSlideBox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showShadow: false,
    };
    this.throttleScroll = throttle(this.onScroll.bind(this), 300);
  }

  componentDidMount() {
    const { show } = this.props;
    if (show) {
      document.body.style.overflow = 'hidden';
    }
  }

  componentWillUnmount() {
    document.body.style.overflow = null;
  }

  onScroll() {
    const { showShadow } = this.state;
    if (this.slideBoxContent && this.slideBoxContainer && this.slideBoxContent.clientHeight > this.slideBoxContainer.clientHeight) {
      const { scrollTop } = this.slideBoxContainer;
      const scrollOnTop = scrollTop === 0;
      if (showShadow && scrollOnTop) {
        this.setState({ showShadow: false });
      } else if (!scrollOnTop && !showShadow) {
        this.setState({ showShadow: true });
      }
    }
  }

  render() {
    const { size, show, header, handleClose, footer, content, className } = this.props;
    const { showShadow } = this.state;
    return (
      show ? (
        <div className={classNames(`${clsPrefix}`, className, { 'show-slidebox': show, 'hide-slidebox': !show, 'show-shadow': showShadow })}>
          <div className={classNames(`${clsPrefix}-container ${size}`)}>
            <div className="slidebox-header">
              <CapHeading type="h1">{header}</CapHeading>
              {<CapIcon onClick={handleClose} type="close" className={classNames(`${clsPrefix}-close-icon`)} />}
            </div>
            <div onScroll={this.throttleScroll} className={classNames('slidebox-content-container', { 'has-footer': footer })} ref={(node) => { this.slideBoxContainer = node; }}>
              <div ref={(node) => { this.slideBoxContent = node; }} style={{ height: '100%' }}>
                {content}
              </div>
            </div>
            {footer && (
              <div className="slidebox-footer">
                {footer}
              </div>
            )}
          </div>
        </div>
      ) : null
    );
  }
}

CapSlideBox.defaultProps = {
  size: "size-r",
};

CapSlideBox.propTypes = {
  header: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.string,
  ]).isRequired,
  footer: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.string,
  ]),
  content: PropTypes.element.isRequired,
  show: PropTypes.bool,
  handleClose: PropTypes.func.isRequired,
  size: PropTypes.string,
  className: PropTypes.string,
};
