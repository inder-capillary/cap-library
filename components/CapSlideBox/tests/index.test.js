import renderer from 'react-test-renderer';
import React from 'react'; // eslint-disable-line no-unused-vars
import CapSlideBox from '../index';

describe('<CapSlider />', () => {
  it('renders correctly', () => {
    const closeSlideBox = jest.fn();
    const tree = renderer.create(
      <CapSlideBox
        show
        size="size-r"
        header="Slide Box Example"
        content={<div>{}</div>}
        handleClose={closeSlideBox}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
