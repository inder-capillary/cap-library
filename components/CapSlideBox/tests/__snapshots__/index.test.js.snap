// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CapSlider /> renders correctly 1`] = `
<div
  className="cap-slide-box-v2 show-slidebox"
>
  <div
    className="cap-slide-box-v2-container size-r"
  >
    <div
      className="slidebox-header"
    >
      <div
        className="sc-bdVaJa cqhfaE"
        type="h1"
      >
        Slide Box Example
      </div>
      <img
        alt=""
        className="cap-slide-box-v2-close-icon"
        onClick={[MockFunction]}
        src="close.svg"
      />
    </div>
    <div
      className="slidebox-content-container"
      onScroll={[Function]}
    >
      <div>
        <div />
      </div>
    </div>
  </div>
</div>
`;
